### ISSUE TITLE
Searching won't work and it is showing up to many broken image content. and picking the right offline browser android app for my tablet?

### ISSUE THREAD

**Issue by KaineRecycler** [gaycodegal/OfflineZipBrowserAndroid#1]
Hi my name is Kaine Dube and I just want report a bug about using the app that you had made for users to view downloaded webpages and view them offline without Internet connection but there was a two major bugs that are not completely working so supposedly this offline browser is offline but it needs to have searching and not show too many broken image content that are found within the offline webpages which I have to report to you on GitHub see if you could fix the searching and broken image content that is in the offline browser android app but yes when I was going to have great experience with my offline webpages which I have been using so first of all the navigation within the website does work and reloading the page does but the searching does not work for my example so and the icons images pictures are not showing up from within the offline website so that's when I had sent you a bug report to your offline browser page on GitHub to see if you could fix and diagnose the apps issue and ask me if there are better apk versions that you have in your release tag page on GitHub because I am looking for a right offline browser that supports my needs that you have so that i can use the right one that has a built in flash player so that it can work on offline webpages that contain with flash content and that is all for right now please comment back to me if you have any questions about it and fixing the bug in your app.

**Comment by gaycodegal** [comment]
Your sentence was very long so I had some trouble understanding exactly what you want. Bullet points would be appreciated. But I'm getting:
- image selection could be improved. 
    - Yes, this should be easy, I can do it but it would help to know what website you cached, and what type of images do you count as broken. 
    - Do you mean images are too small, images are transparent, or they are broken in some other way. Post a screenshot if you feel comfortable to show me
- searching: not sure what you mean by this. I need more info to help. It may be specific to your use case.
    - if you are using an app offline that has search powered by a server, it will not work offline, because it will never contact the server. Apps loaded here are literally unable to contact other computers or wifi.
- flash: impossible. I would love to, but it's not feasible. Now technically if you had a browser that provides a web view set as your default browser that supports flash, it's maybe possible. But I know of no such app

**Comment by KaineRecycler** [comment]
But could you please be very helpful and try to enable searching for your app that you have on GitHub please maybe you could install a offline search server for your offline browser and that would be a helpful tool for other users to search for content from within a offline website only if you could install a offline search server for your app but I would have rather have recommended a search server called elastic search search server for offline searching and that would highly recommend helping out users to search and browse and navigate offline like a typical offline zip browser but anyway I would like to your app to use elastic search server for offline searching and browsing please just let me know when you can look into it so that I can browse websites offline without Internet connection and then once when you have fix that bug report before I can start recommending a flash player for your app that I had found on GitHub that was not made to long ago.

**Comment by gaycodegal** [comment]
I could do that, although not with elastic search, but I'm not going to.

It's possible to make a zip file support searching itself without any additional capabilities from the host Android app, and it's not a feature I personally need so I'm not going to add it.

I may eventually add better history management and navigation, but search is not a priority for now. 

This app was originally designed so I could play text based games that often are packaged as zipped html. I mainly open sourced it and made it available to others because I couldn't find anything else that did this.

You are welcome to fork the repository and make it support searching/flash, but they don't sound like features I'm interested in developing at the moment.

Search is more of a feature that would benefit "I downloaded Wikipedia" and it's a hard problem to solve. Search typically involves a whole bunch of pre-computation that can make for very large files, or it will be extremely slow. And it's site-specific what should or shouldn't be searchable. Basically it's a very hard problem and not something easy to add to an app.

Searching text displayed currently on screen is easy, searching text in files that are not currently loaded is hard.

But anyways, if you want to do this hard work yourself you're welcome to fork the repository, but the feature would make the app too large, or be buggy, so it's not something I plan on adding.

Similarly, with flash; even if there were GitHub repositories with Android flash code, the codebase of flash is massive and not a good fit for this app, it would be better to have flash live in its own browser and then have that browser provide the default web view, which is what this app consumes.

So TLDR: No to flash, no to searching entire zip files, maybe to the icon thing (still unsure of what's wrong, but I do try to sort the potential icons by quality, and I can adjust the quality ordering)

**Comment by gaycodegal** [comment]
On the icon front for instance, I just sort them by how long the path is (how many directories it's in specifically). Perhaps I should sort for icons named things like icon.png

Current algorithm (its basic):
https://github.com/gaycodegal/OfflineZipBrowserAndroid/blob/aa72ad2fe3e7e9c0d2a2c614fcc3ddd07ccac553/app/src/main/java/foss/zip/offline/browser/offlinezipbrowser/ShortLinkMakerActivity.kt#L159
